### ISSUE TITLE
feat: expose Newton Tikhonov damping as a user-configurable parameter

### ISSUE THREAD

**Issue by kovacoj** [kovacoj1/optimizers#67]
## Problem

Newton hardcodes a `1e-4` Tikhonov damping term:

```python
H += 1e-4 * torch.eye(self.numel, ...)
```

For a numerical mathematician, this is as fundamental as `mu` is for Levenberg-Marquardt. The damping strength determines:
- How much the Newton direction is regularized toward gradient descent.
- Whether the method can handle singular or near-singular Hessians.
- The convergence rate on ill-conditioned problems.

A user who needs stronger or weaker regularization has no way to control it. The `1e-4` value is not documented, and there's no way to disable it entirely (e.g., for problems where the Hessian is known to be PD).

## Proposal

Add a `damping` parameter (default `1e-4` for backward compatibility):

```python
class Newton(torch.optim.Optimizer):
    def __init__(self, params, line_search_method=None, damping=1e-4):
        ...
```

This follows the same pattern as `LevenbergMarquardt(mu=...)` and gives users full control over the regularization strength.

**Comment by kovacoj** [comment]
## Proposed fix

Add a `damping` parameter to Newton's constructor, defaulting to `1e-4` for backward compatibility:

```python
class Newton(torch.optim.Optimizer):
    def __init__(self, params, line_search_method=None, damping=1e-4):
        super().__init__(params, dict(line_search_method=line_search_method, damping=damping))
        # ...
        self.damping = damping
```

Then add the standard property/setter pair:

```python
@property
def damping(self):
    return self.param_groups[0]['damping']

@damping.setter
def damping(self, value):
    self.param_groups[0]['damping'] = value
```

And use it in `step()`:

```python
# Before
H += 1e-4 * torch.eye(self.numel, device=prototype.device, dtype=prototype.dtype)

# After
H += self.damping * torch.eye(self.numel, device=prototype.device, dtype=prototype.dtype)
```

**Why `damping` and not `lambda` or `alpha`:**
- `damping` is the standard term in the numerical optimization literature for Tikhonov regularization.
- `lambda` is a Python reserved word.
- `alpha` is already used for step sizes in line search.
- Matches the naming convention of `mu` in LM (another damping parameter).

**Setting `damping=0`** disables the regularization entirely, which is valid when the Hessian is known to be positive-definite.

**Serialization:** Since `damping` is stored in `param_groups`, it's automatically included in `state_dict()` / `load_state_dict()` — no extra work needed.
